diff -ruN .gitignore .gitignore
--- .gitignore	1969-12-31 16:00:00.000000000 -0800
+++ .gitignore	2022-08-05 09:51:07.000000000 -0700
@@ -0,0 +1,2 @@
+*
+!.gitignore
diff -ruN registries/PoolRegistry.sol registries/PoolRegistry.sol
--- registries/PoolRegistry.sol	2022-08-05 08:23:59.000000000 -0700
+++ registries/PoolRegistry.sol	2022-08-05 09:51:07.000000000 -0700
@@ -42,8 +42,8 @@
     /// @inheritdoc IPoolRegistry
     mapping(bytes32 => uint256) public override poolsIndexOfNamePlusOne;
 
-    PoolData[] private pools;
-    mapping(uint256 => address[]) private eligiblePairsMap;
+    PoolData[] public pools;                                // MUNGED private => public
+    mapping(uint256 => address[]) internal eligiblePairsMap; // MUNGED private => internal
 
     /**
      * @notice Add a new registry entry to the master list.
@@ -185,7 +185,7 @@
             : _getSwapStorage(inputData.poolAddress).lpToken;
 
         // Check token addresses
-        for (uint8 i = 0; i < 8; i++) {
+        for (uint8 i = 0; i < getConstantLength(); i++) {
             try ISwap(inputData.poolAddress).getToken(i) returns (
                 IERC20 token
             ) {
@@ -561,4 +561,8 @@
                 .div(baseLPTokenTotalSupply);
         }
     }
+
+    function getConstantLength() public view returns (uint256) {
+        return 8; 
+    }
 }
